include/seastar/rpc/rpc.hh

```diff
@@ -383,6 +385,23 @@ public:
         _outgoing_queue_ready = dummy->done.get_future();
         (void)p.get_future().then([dummy = std::move(dummy)] { dummy->done.set_value(); });
     }
+
+    // Each RPC connection has one such worker fiber, whose sole purpose
+    // is to eventually send an empty RPC frame over the connection whenever
+    // the compressor demands it (by signalling a condition variable).
+    //
+    // This is a mechanism which allows the compressor to communicate
+    // something to the other end of the connection (e.g. to re-negotiate
+    // some parameters of compression, such as a dictionary).
+    future<> spawn_compressor_bumper() {
+        return seastar::repeat([this] {
+            return _compressor_needs_progress.wait().then([this] () {
+                return send({})
```

Comment:
I do not see the usage in the series, but why can't whoever suppose to signal  `_compressor_needs_progress` simply call `send({})`?

Reply:
The purpose of all this is to let the compressor send some information over the connection. So `_compressor_needs_progress` is supposed to be signaled by compressors.

So the problems with calling `send({})` directly are:
1. `send({})` is a private interface of `connection`.
2. Compressors are synchronous. If something calls `send({})`, it should await the future, but compressors have no way to await anything.

You could shuffle things around (e.g. add `future<> close();` to compressors and await it when closing a connection) to make calling `send({})` directly possible, but what you get at the end is going to be an equivalent of this variant with `condition_variable`.

Reply:
So the alternative can be:
1. Making connection compressor's friend
2. Pass the connection to a compressor
3. add `future<> compressor::close()`
4. call the `close()` where we call `_compressor_needs_progress.broken()` now

I am not sure this is worse than fiber (which we start regardless if  compression is enabled or not?). @avikivity ?

Reply:
Yes, that's what the alternative would look like. They are pretty much equivalent.

The indirect variant with `condition_variable` (or, say, `std::function`) has the benefit of being easier to write tests with, though. It allows you to create and test compressors without dealing with `connection`s.

Reply:
I, personally, would prefer `std::function` + `future<> compressor::close()` to a fiber.

Reply:
@avikivity Do you have an opinion about this?

Reply:
I agree that std::function is nicer rather than exposing an internal.

Reply:
> I agree that std::function is nicer rather than exposing an internal.

That we all agree with, but that wasn't the question.

Anyway, to avoid dragging the discussion, I'm going to follow Gleb's preference and move the fiber out of `connection`.